src/App.vue

```diff
@@ -93,8 +93,17 @@ export default {
         })(),
         (async () => {
           const tokens = await Backend.getTokenBalances(this.address);
-          await Promise.all(Object.entries(tokens).map(async ([token]) => this
-            .addTokenBalance({ token, balance: await tokenBalance(token, this.address) })));
+          await Promise.all(Object.entries(tokens).map(async ([token]) => {
```

Comment:
how do you mean?

Reply:
currently none of the backend call resulting data is persisted, which should be for the overall logic, just moving the call to a different store doesn't help

also as we rely on data from aeternity as well I don't think it should be in store

Reply:
> currently none of the backend call resulting data is persisted, which should be for the overall logic, just moving the call to a different store doesn't help

Firstly, App is not a good place to store a blob of initialization logic that is needed for other components. It is ok to trigger this initialization from here. Probably the best could be to do a self-contained plugin for the store that will initialize itself when it is called by the store.

The question is not in persisting, I'm proposing to keep together the definition of the state, it's mutations, and actions (like fetching of token data), and to clean up App at the same time.

> also as we rely on data from aeternity as well I don't think it should be in store

In general, we should have
- a set of components depending on its internal state and vuex
- internal state and vuex be mutated by methods or mutations called by actions
- helper functions that may work with some API or do utility functions, but they shouldn't depend on the vuex

I think the whole "aeternity.js" should be in vuex. Because
- it is the common state and methods that are needed for multiple components (can't be a part of a single component)
- it depends on the store's state (sdk, current network), so makes the explicit dependency on the store.

Making the explicit dependency between components and the store:
- complicates unit testing of components (more difficult to stub the store)
- complicates SSR (the component become dependant on a single instance of a store that may be multiple in a single server thread)

Check 5ae9ce28a98cb28600b18176db3b47d721de8dc8 for the way how I'm proposing to move tokens data fetching to store.